c/array/array_rotate.c

```diff
@@ -0,0 +1,62 @@
+
+#include <stdbool.h>
+#include <assert.h>
+#include <time.h>
+#include "array_utils.h"
+
+#define MAX_SIZE 10
+
+/*
+ *rotate array rotaates the given original array for the given
+ *number of times.
+ *for example 1 2 3 4 5 6 7 8 9 10 is the original array
+ *then it roates it to 9 10 1 2 3 4 5 6 7 8 when number of times
+ *rotate is 2.
+*/
+
+void rotate_array(int *nums, int size, int rotate_count)
+{
+    for (int index = 0; index < rotate_count; index++)
+    {
+        int last_element = nums[size - 1];
+        for (int i = size - 1; i > 0; i--)
+        {
+            nums[i] = nums[i - 1];
+        }
+        nums[0] = last_element;
+    }
+}
+
+void reverse_array_for_rotate(int *nums,int size)
```

Comment:
Why a seperate function for reverse array here?
Why don't you just call the "reverse_array" function itself from here?